src/olympia/addons/tests/test_admin.py

```diff
@@ -177,6 +177,32 @@ def test_can_not_edit_without_addons_edit_permission(self):
         addon.reload()
         assert addon.guid == '@foo'
 
+    def test_access_using_slug(self):
+        addon = addon_factory(guid='@foo')
+        self.detail_url = reverse(
+            'admin:addons_addon_change', args=(addon.slug,)
+        )
+        user = user_factory()
+        self.grant_permission(user, 'Admin:Tools')
+        self.grant_permission(user, 'Addons:Edit')
+        self.client.login(email=user.email)
+        response = self.client.get(self.detail_url, follow=True)
```

Comment:
I think I'd rather see a test proving we redirect to the right URL (so `folllow=False`, you can use our `assert3xx` method with the expected URL). Same thing below.